src/material-experimental/mdc-progress-spinner/progress-spinner.ts

```diff
@@ -0,0 +1,268 @@
+    super(_elementRef);
+    this._noopAnimations = animationMode === 'NoopAnimations' &&
+      (!!defaults && !defaults._forceAnimations);
+
+    if (defaults) {
+      if (defaults.diameter) {
+        this.diameter = defaults.diameter;
+      }
+
+      if (defaults.strokeWidth) {
+        this.strokeWidth = defaults.strokeWidth;
+      }
+    }
+  }
+
+  private _mode: ProgressSpinnerMode = 'determinate';
+
+  /**
+   * Mode of the progress bar.
+   *
+   * Input must be one of these values: determinate, indeterminate, buffer, query, defaults to
+   * 'determinate'.
+   * Mirrored to mode attribute.
+   */
+  @Input()
+  get mode(): ProgressSpinnerMode {
+    return this._mode;
+  }
+
+  set mode(value: ProgressSpinnerMode) {
+    // Note that we don't technically need a getter and a setter here,
+    // but we use it to match the behavior of the existing mat-progress-bar.
+    this._mode = value;
+    this._syncFoundation();
+  }
+
+  private _value = 0;
+
+  /** Value of the progress bar. Defaults to zero. Mirrored to aria-valuenow. */
+  @Input()
+  get value(): number {
+    return this.mode === 'determinate' ? this._value : 0;
+  }
+
+  set value(v: number) {
+    this._value = Math.max(0, Math.min(100, coerceNumberProperty(v)));
+    this._syncFoundation();
+  }
+
+  private _diameter = BASE_SIZE;
+
+  /** The diameter of the progress spinner (will set width and height of svg). */
+  @Input()
+  get diameter(): number {
+    return this._diameter;
+  }
+
+  set diameter(size: number) {
+    this._diameter = size;
+    this._syncFoundation();
+  }
+
+  private _strokeWidth: number;
+
+  /** Stroke width of the progress spinner. */
+  @Input()
+  get strokeWidth(): number {
+    return this._strokeWidth || this.diameter / 10;
+  }
+
+  set strokeWidth(value: number) {
+    this._strokeWidth = coerceNumberProperty(value);
+  }
+
+  /** The radius of the spinner, adjusted for stroke width. */
+  get _circleRadius() {
+    return (this.diameter - BASE_STROKE_WIDTH) / 2;
+  }
+
+  /** The view box of the spinner's svg element. */
+  get _viewBox() {
+    const viewBox = this._circleRadius * 2 + this.strokeWidth;
+    return `0 0 ${viewBox} ${viewBox}`;
+  }
+
+  /** The stroke circumference of the svg circle. */
+  get _strokeCircumference(): number {
+    return 2 * Math.PI * this._circleRadius;
+  }
+
+  /** The dash offset of the svg circle. */
+  get _strokeDashOffset() {
+    if (this.mode === 'determinate') {
+      return this._strokeCircumference * (100 - this._value) / 100;
+    }
+
+    return null;
+  }
+
+  /** Stroke width of the circle in percent. */
+  get _circleStrokeWidth() {
+    return this.strokeWidth / this.diameter * 100;
+  }
+
+  ngAfterViewInit() {
+    const element = this._elementRef.nativeElement;
+
+    this._rootElement = element.querySelector('.mdc-circular-progress') as HTMLElement;
+    this._determinateCircle =
+      element.querySelector('.mdc-circular-progress__determinate-container') as HTMLElement;
```

Comment:
Instead of `querySelector` here, you can use a `ViewChild` which is a bit more idiomatic.